### DISCUSSION TITLE
Word autocomplete

### DISCUSSION THREAD

**Discussion by ghost** [orbitalquark/textadept#372] [Q&A] [answered]
It's not working out of the box. It's a normal text file so it's not related to language servers. First I thought autocomplete is disabled so I had to use shortcut keys to manually trigger it. I pressed `Ctrl+Space` but nothing happened. Then I found on Edit menu the shortcut key for word autocomplete on TextAdept is `Ctrl+Return`. To my limited knowledge, I think `Return` is `Enter`. So I tried `Ctrl+Enter`, it also didn't work!

p/s: Could you tell me why you decided to use `Ctrl+Return` for word autocomplete on TextAdept instead of `Ctrl+Space`? It's always `Ctrl+Space` on other text editors, or `Ctrl+Shift+Space`. Btw, what does `Ctrl+Space` do on TextAdept?

**Comment by orbitalquark** [comment]
I opened a new buffer, typed "foo" followed by Enter (new line). Then I 
typed "f" followed by Ctrl+Enter. It completed "oo" so now I have two 
lines, both containing "foo". Does using the Edit > Complete Word menu 
item work? Maybe your key binding is not being recognized for some reason.

**Reply by ghost** [reply]
Is there a way to configure it to automatically trigger without the need of typing `Ctrl+Enter` like on Lite XL? Btw, could I configure it to use another shortcut key than `Ctrl+Enter`? Thank you.

**Comment by orbitalquark** [comment]
Yes, and yes. The manual has information on setting up key bindings. For 
the former, you can hook into the `events.CHAR_ADDED` event. I believe 
there was a previous discussion on this topic.

**Reply by ghost** [reply]
Believe me or not, I don't know how to write even a single line of Lua code. LPeg is simple enough for me could chimed in to make lexers. It's all that I can do. Even a basic functionality like this that should work out of the box needs you to implement it yourself. I'm really disappointed.

**Comment by orbitalquark** [comment, marked as answer]
Sorry to disappoint. I wrote Textadept for myself and decided to share 
it with you all. I don't need this feature, so I didn't implement it. I 
gave users that want it the option to implement it themselves. If this 
isn't your thing, then you may want to consider a more full-featured editor.

**Reply by ghost** [reply]
https://github.com/orbitalquark/textadept/pull/376
